File: src/main/java/cube8540/oauth/authentication/credentials/oauth/scope/application/OAuth2ScopeManagementService.java
package cube8540.oauth.authentication.credentials.oauth.scope.application;

import cube8540.oauth.authentication.credentials.AuthorityDetails;
import cube8540.oauth.authentication.credentials.AuthorityDetailsService;

import java.util.Collection;

public interface OAuth2ScopeManagementService extends AuthorityDetailsService {

    Long countByScopeId(String scopeId);

    Collection<AuthorityDetails> loadAllScopes();

    Collection<AuthorityDetails> loadPublicScopes();

    AuthorityDetails registerNewScope(OAuth2ScopeRegisterRequest registerRequest);

    AuthorityDetails modifyScope(String scopeId, OAuth2ScopeModifyRequest modifyRequest);

    AuthorityDetails removeScope(String scopeId);

}

File: src/main/kotlin/cube8540/oauth/authentication/oauth/scope/application/OAuth2ScopeManagementService.kt
package cube8540.oauth.authentication.oauth.scope.application

interface OAuth2ScopeManagementService {

    fun countByScopeId(scopeId: String): Long

    fun registerNewScope(registerRequest: OAuth2ScopeRegisterRequest): OAuth2ScopeDetails

    fun modifyScope(scopeId: String, modifyRequest: OAuth2ScopeModifyRequest): OAuth2ScopeDetails

    fun removeScope(scopeId: String): OAuth2ScopeDetails

}